### PR TITLE
Octopus Deploy Tentacle VM Extension template

### PR DESCRIPTION
**Pull request by matt-richardson**
Initial draft

### CODE REVIEW

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      "defaultValue": "Standard_D2_v2",
+      "type": "string",
+      "metadata": {
+        "description": "Size of the Virtual Machine"
+      }
+    },
+    "tentacleOctopusServerUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "The URL of the octopus server with which to register"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: Capitalise Octopus Server

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "vmSize": {
+      "defaultValue": "Standard_D2_v2",
+      "type": "string",
+      "metadata": {
+        "description": "Size of the Virtual Machine"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: Some descriptions use `.` some don't

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    "tentacleOctopusServerUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "The URL of the octopus server with which to register"
+      }
+    },
+    "tentacleApiKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The Api Key to use to register the Tentacle with the server"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: "with the Octopus Server"

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+    },
+    "tentacleCommunicationMode": {
+      "defaultValue": "Listen",
+      "allowedValues": [
+        "Listen",
+        "Poll"
+      ],
+      "type": "string",
+      "metadata": {
+        "description": "The type of Tentacle - whether the Tentacle listens for requests from server, or actively polls the server for requests"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: server => Octopus Server

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+      }
+    },
+    "tentaclePort": {
+      "defaultValue": 10933,
+      "minValue": 0,
+      "maxValue": 65535,
+      "type": "int",
+      "metadata": {
+        "description": "The port on which the Tentacle should listen, when CommunicationMode is set to Listen, or the port on which to poll the server, when CommunicationMode is set to Poll. By default, Tentacle's listen on 10933 and poll the Octopus Server on 10943."
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: server => Octopus Server

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+      }
+    },
+    "tentacleRoles": {
+      "type": "string",
+      "metadata": {
+        "description": "A comma delimited list of roles to apply to the Tentacle"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: Capitalise Roles in this case I think. Can you provide links to our docs? (Probably not)

File: 101-octopus-deploy-tentacle-windows/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+      }
+    },
+    "tentacleEnvironments": {
+      "type": "string",
+      "metadata": {
+        "description": "A comma delimited list of environments in which the Tentacle should be placed"
```
Comment (michaelnoonan) [inline 101-octopus-deploy-tentacle-windows/azuredeploy.json]: Same comment as Roles above.

**Comment by matt-richardson** [comment]
cc @paulbouwer

**Comment by singhkays** [comment]
@matt-richardson Please rename the template folder to octopus-deploy-tentacle-windows since this is an application template

**Comment by matt-richardson** [comment]
Folder renamed. Any other changes needed?

**Comment by singhkays** [comment]
merged

**Pull request merged**